smt/surrogate_models/krg_based.py

```diff
@@ -187,24 +234,27 @@ def _reduced_likelihood_function(self, theta):
         reduced_likelihood_function_value = -np.inf
         par = {}
         # Set up R
-        MACHINE_EPSILON = np.finfo(np.double).eps
-        nugget = 10.0 * MACHINE_EPSILON
-        if self.name == "MFK":
-            if self._lvl != self.nlvl:
-                # in the case of multi-fidelity optimization
-                # it is very probable that lower-fidelity correlation matrix
-                # becomes ill-conditionned
-                nugget = 10.0 * nugget
-        elif self.name in ["MGP"]:
-            nugget = 100.0 * nugget
+        nugget = self.options["nugget"]
+        if self.options["eval_noise"]:
+            nugget = 0
```

Comment:
Why nugget= 0 ? We want to allow both nugget and noise usage.

Reply:
There is no need if there is a noise... The noise will play the role of the nugget

Reply:
The nugget is for numerical problems. it is homoscedastic and fixed.
The noise is for noisy data, it could be both homoscedastic or heteroscedastic.

Evaluating a noise to play the role of the nugget is dirty. It could word but it is not a good way to do it.

Reply:
Eval_noise should evaluate the noise, not the sum "nugget+noise" sometimes and just "noise" otherwise

Reply:
No, we are not going to use the noise as a nugget... The nugget has to be considered only in cases with numerical problems... For the noisy case, we do not need that nugget because there is no such problem because of the influence of the noise.

Reply:
Can you at least, set the default nugget to 0 and remove the "set to 0" with eval_noise.

Reply:
- "nugget", 0, types=(float, list), desc="a jitter for numerical stability" 
.....
- if not self.options["eval_noise"]:
                MACHINE_EPSILON = np.finfo(np.double).eps
                nugget = 1e2 * MACHINE_EPSILON

Reply:
Another question, why "list" type for nugget ?